tests/testthat/test-target_sda.R

```diff
@@ -20,6 +20,25 @@ test_that("with fake data outputs known value", {
   )
 
   expect_known_value(out, "ref-target_sda", update = FALSE)
+
+  out_company <- target_sda(
+    fake_matched(
+      sector_ald = "cement"
+    ),
+    ald = fake_ald(
+      sector = "cement",
+      technology = "cement",
+      year = c(2020, 2021, 2022),
+      emission_factor = c(1, 2, 3)
+    ),
+    co2_intensity_scenario = fake_co2_scenario(
+      year = c(2020, 2050),
+      emission_factor = c(0.6, 0.2)
+    ),
+    by_company = TRUE
+  )
+
+  expect_known_value(out_company, "ref-target_sda_company", update = TRUE)
```

Comment:
Some of the refactoring caused the checks to fail when generating the examples. I added a test to help with this.

Reply:
Is that failure the reason to use `expect_known_value()` instead of `expect_snapshot()`?

Reply:
No, I only use `expect_known_value()` as this is what the rest of the package seems to be using still. Maybe in a separate refactoring we can switch everything to `expet_snapshot()`?

Reply:
Thanks! The issue with separate refactoring is that it rarely happens. "Cleaning as you go" tends to keep the house cleaner than "I'll do it on Saturday"

Reply:
Sure, I guess my hope was to keep this PR relatively focused (I even seem to have already failed that). Changing the `expect_known_value()` to `expect_snapshot()` in every test that it occurs in, seems like it is completely unrelated to this PR and it's purpose. 

Also, I was very hesitant to touch the tests at all in this PR, as the purpose is refactoring and I wanted to ensure the output was consistent with before. This was (hopefully) the only place that I touched a test, as I was actually adding a test where we were lacking coverage.